lib/loader/serialize/serial_instruction.cpp

```diff
@@ -76,36 +76,64 @@ Serializer::serializeInstruction(const AST::Instruction &Instr,
     assumingUnreachable();
   }
 
+  auto serializeBlockType = [this,
+                             &OutVec](const BlockType &Type) -> Expect<void> {
+    if (Type.isEmpty()) {
+      OutVec.push_back(static_cast<uint8_t>(TypeCode::Epsilon));
+    } else if (Type.isValType()) {
+      if (auto Res = serializeValType(Type.getValType(),
+                                      ASTNodeAttr::Instruction, OutVec);
+          unlikely(!Res)) {
+        return Unexpect(Res);
+      }
+    } else {
+      if (unlikely(!Conf.hasProposal(Proposal::MultiValue))) {
+        return logNeedProposal(ErrCode::Value::MalformedValType,
+                               Proposal::MultiValue, ASTNodeAttr::Instruction);
+      }
+      serializeS33(static_cast<int64_t>(Type.getTypeIndex()), OutVec);
+    }
+    return {};
+  };
+
   // Serialize immediate.
   switch (Instr.getOpCode()) {
   // Control instructions.
   case OpCode::Unreachable:
   case OpCode::Nop:
   case OpCode::Return:
+  case OpCode::Throw_ref:
   case OpCode::End:
   case OpCode::Else:
     return {};
 
   case OpCode::Block:
   case OpCode::Loop:
   case OpCode::If:
-    if (Instr.getBlockType().isEmpty()) {
-      OutVec.push_back(static_cast<uint8_t>(TypeCode::Epsilon));
-    } else if (Instr.getBlockType().isValType()) {
-      if (auto Res = serializeValType(Instr.getBlockType().getValType(),
-                                      ASTNodeAttr::Instruction, OutVec);
-          unlikely(!Res)) {
-        return Unexpect(Res);
-      }
-    } else {
-      if (unlikely(!Conf.hasProposal(Proposal::MultiValue))) {
-        return logNeedProposal(ErrCode::Value::MalformedValType,
-                               Proposal::MultiValue, ASTNodeAttr::Instruction);
+    return serializeBlockType(Instr.getBlockType());
+
+  case OpCode::Try_table: {
+    // Serialize the result type.
+    if (auto Res = serializeBlockType(Instr.getTryCatch().ResType); !Res) {
+      return Unexpect(Res);
+    }
+    // Serialize the vector of catches.
+    uint32_t VecCnt = static_cast<uint32_t>(Instr.getTryCatch().Catch.size());
+    serializeU32(VecCnt, OutVec);
+    for (auto Catch : Instr.getTryCatch().Catch) {
+      // Read the tag index.
```

Comment:
Lost the catch flag.
The element of the catch clause has a flag (`0x00` to `0x03`) before the tag index and label index.

Reply:
I thought the catch instruction is marked as Legacy EH.

Reply:
I men the catch clause in the `try_table` instruction, not the `catch` instruction.
ref: https://webassembly.github.io/exception-handling/core/binary/instructions.html#control-instructions

Reply:
I think it's written here : https://github.com/WasmEdge/WasmEdge/blob/master/lib/loader/ast/instruction.cpp#L421

Reply:
It says we should remove these flags when removing the deprecated EH.

Reply:
It means the line `Desc.IsLegacy = false;`.
Please follow the binary format of spec ref.

Reply:
The `try table` binary format in spec:
```
0x1F bt:blocktype c*:vec(catch) (in:instr)* 0x0B

catch ::= 0x00 x:tagidx l:labelidx
       | 0x01 x:tagidx l:labelidx
       | 0x02 l:labelidx
       | 0x03 l:labelidx
```
Do you find that `0x00` - `0x03` flag in the catch clause?

Reply:
I am sorry, I missed this. I thiught the the all flags are deprecated. I will update the patch.